SharedSources/VLCMediaLibraryManager.swift

```diff
@@ -151,6 +164,111 @@ class VLCMediaLibraryManager: NSObject {
     func genre(sortingCriteria sort: VLCMLSortingCriteria = .default, desc: Bool = false) -> [VLCMLGenre] {
         return medialib.genres(with: sort, desc: desc)
     }
+
+    func fetchMedia(with mrl: URL?) -> VLCMLMedia? {
+        guard let mrl = mrl  else {
+            return nil
+        }
+        return medialib.media(withMrl: mrl)
+    }
+
+    @discardableResult
+    func prepareMigrationIfNeeded() -> Bool {
+        if !didMigrate {
+            migrationDelegate?.medialibraryDidStartMigration(self)
+            return true
+        }
+        return false
+    }
+}
+
+// MARK: - Migration
+
+private extension VLCMediaLibraryManager {
+    // Ugly hack to get the realPath
+    func realPath(to url: URL) -> URL? {
+        var path = url.absoluteString
+        path = path.replacingOccurrences(of: "file:///var/mobile", with: "file:///private/var/mobile")
+        return URL(string: path)
+    }
+
+    func startMigrationIfNeeded() {
+        guard !didMigrate else {
+            return
+        }
+
+        guard migrateToNewMediaLibrary() else {
+            migrationDelegate?.medialibraryDidStopMigration(self)
+            return
+        }
+
+        migrationDelegate?.medialibraryDidFinishMigration(self)
+    }
+
+    func migrateMedia(_ oldMedialibrary: MLMediaLibrary) -> Bool {
+        guard let allFiles = MLFile.allFiles() as? [MLFile] else {
+            assertionFailure("VLCMediaLibraryManager: Migration: Unable to retreive all files")
+            return false
+        }
+
+        for media in allFiles {
+            if let newMedia = fetchMedia(with: realPath(to: media.url)) {
+                newMedia.updateTitle(media.title)
+                newMedia.setPlayCount(media.playCount.uint32Value)
```

Comment:
should we also set the Metadata field "Seen" here ?

Reply:
Yes, we definitely could, but it's should be more or less implicit with the play count if it's correctly set.

Reply:
you're right, that it is implicit but we will forget that at some point and just check for for the isSeen flag. For the new tag for example and we claim this media was never seen so it would show it.

Reply:
Sorry, I planned on doing it but I completely forgot about it! It should be all good now. 👍